### PR TITLE
Remove references to complement's `master` branch

### PR DESCRIPTION
**Pull request by richvdh**
I'd like us to consider renaming to `main`, so let's remove the references to `master` specifically.

### CODE REVIEW

File: dendrite/pipeline.yml [hunk 1]
```diff
@@ -38,14 +38,14 @@ steps:
 
   - command:
       - docker build -t complement-dendrite -f build/scripts/Complement.Dockerfile .
-      - "wget -N https://github.com/matrix-org/complement/archive/master.tar.gz && tar -xzf master.tar.gz"
-      - "cd complement-master && COMPLEMENT_BASE_IMAGE=complement-dendrite:latest go test -v -tags=\"dendrite_blacklist\" ./tests"
+      - "wget -O - https://github.com/matrix-org/complement/archive/HEAD.tar.gz | tar -xz --strip-components=1 -C complement"
```
Comment (richvdh) [inline dendrite/pipeline.yml:42]: the `--strip-components=1 -C complement` means that we don't care about whether the folder in the tarball is named `complement-master` or named after whatever `HEAD` happens to point to right now.

File: synapse/pipeline.yml [hunk 2]
```diff
@@ -234,8 +234,8 @@ steps:
     plugins:
       - matrix-org/download#v1.1.0:
           urls:
-            - https://raw.githubusercontent.com/matrix-org/pipelines/master/synapse/docker-compose.yaml
-            - https://raw.githubusercontent.com/matrix-org/pipelines/master/synapse/docker-compose-env
+            - https://raw.githubusercontent.com/matrix-org/pipelines/HEAD/synapse/docker-compose.yaml
```
Comment (richvdh) [inline synapse/pipeline.yml:237]: thought I might as well do these while I was here.

File: dendrite/pipeline.yml (hunk 1 above)
Comment (anoadragon453) [reply dendrite/pipeline.yml:42]: FYI you'll need to `mkdir complement` as `tar` expects the directory to exist before it writes into it.

**Review by anoadragon453 — DISMISSED** [review]
lgtm besides the one change below. TIL one can just pop `HEAD` in there though!

**Pull request merged**